src/library/features/crates/cratefeaturehelper.cpp

```diff
@@ -74,6 +74,68 @@ CrateId CrateFeatureHelper::createEmptyCrate() {
     if (m_pTrackCollection->insertCrate(newCrate, &newCrateId)) {
         DEBUG_ASSERT(newCrateId.isValid());
         newCrate.setId(newCrateId);
+        m_pTrackCollection->crates().initClosureForCrate(newCrateId);
+        m_pTrackCollection->crates().generateCratePaths(newCrate);
```

Comment:
Calling TrackCollection::insertCrate above should take care of this. TrackCollection shouldn't have to be concerned with these details either. How about moving this logic to CrateStorage?

CrateFeature does not make use of the returned CrateId from CrateFeatureHelper, so you can change the return type to void.